src/drt/src/io/io.cpp

```diff
@@ -146,7 +146,40 @@ void io::Parser::setInsts(odb::dbBlock* block)
     tmpInst->setOrient(inst->getOrient());
     int numInstTerms = 0;
     tmpInst->setPinAccessIdx(inst->getPinAccessIdx());
+    dbTransform xform = tmpInst->getUpdatedXform();
+    auto grid = tech_->getManufacturingGrid();
     for (auto& uTerm : tmpInst->getMaster()->getTerms()) {
+      if (master->getMasterType().isBlock()) {
+        for (auto& pin : uTerm->getPins()) {
+          for (auto& uFig : pin->getFigs()) {
+            if (uFig->typeId() == frcRect) {
+              auto shape = uFig.get();
+              Rect box = shape->getBBox();
+              xform.apply(box);
+              if (box.xMin() % grid || box.yMin() % grid || box.xMax() % grid
+                  || box.yMax() % grid) {
+                logger_->error(DRT,
+                               416,
+                               "Term {} of {} contains offgrid pin shape",
```

Comment:
Maybe the message can be more specific? Like
"Term {} of {} contains an off-grid pin shape. Pin shape ({} {}, {} {}) is not a multiple of the manufacturing grid {}."